src/pybind/mgr/volumes/fs/operations/versions/subvolume_v2.py

```diff
@@ -375,13 +376,21 @@ def remove(self, retainsnaps=False, internal_cleanup=False):
                 self.auth_mdata_mgr.delete_subvolume_metadata_file(self.group.groupname, self.subvolname)
                 return
         if self.state != SubvolumeStates.STATE_RETAINED:
-            self.trash_incarnation_dir()
-            self.metadata_mgr.remove_section(MetadataManager.USER_METADATA_SECTION)
-            self.metadata_mgr.update_global_section(MetadataManager.GLOBAL_META_KEY_PATH, "")
-            self.metadata_mgr.update_global_section(MetadataManager.GLOBAL_META_KEY_STATE, SubvolumeStates.STATE_RETAINED.value)
-            self.metadata_mgr.flush()
-            # Delete the volume meta file, if it's not already deleted
-            self.auth_mdata_mgr.delete_subvolume_metadata_file(self.group.groupname, self.subvolname)
+            try:
+                self.metadata_mgr.update_global_section(MetadataManager.GLOBAL_META_KEY_PATH, "")
+                self.metadata_mgr.update_global_section(MetadataManager.GLOBAL_META_KEY_STATE, SubvolumeStates.STATE_RETAINED.value)
+                self.metadata_mgr.remove_section(MetadataManager.USER_METADATA_SECTION)
+                self.metadata_mgr.flush()
+                self.trash_incarnation_dir(subvol_path)
+                # Delete the volume meta file, if it's not already deleted
+                self.auth_mdata_mgr.delete_subvolume_metadata_file(self.group.groupname, self.subvolname)
+            except cephfs.Error as e:
+                if e.errno == errno.ENOSPC:
+                    raise VolumeException(-errno.ENOSPC, "No space left")
+            except MetadataMgrException as e:
+                log.error(f"failed to write config: {e}")
+                raise VolumeException(-e.args[0], e.args[1])
```

Comment:
This is the bug that was misleading us and causing the test failure of http://qa-proxy.ceph.com/teuthology/neesingh-2025-08-13_06:04:47-fs-wip-neesingh-testing1-130825-distro-default-smithi/8440209/teuthology.log

The final error return value should be negative. The `MetadataMgrException` has the negative error properly filled. You are converting it to a positive return value while raising `VolumeException`. Hence, you are getting bash return value as 0 with the error string.

Do the following change a try.

```raise VolumeException(e.args[0], e.args[1])```

Reply:
Thanks for looking into it,Will update